pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1557,6 +1558,7 @@ public void uploadToDeepStoreIfMissing(TableConfig tableConfig, List<SegmentZKMe
 
           // Randomly ask one server to upload
           URI uri = peerSegmentURIs.get(RANDOM.nextInt(peerSegmentURIs.size()));
+          String crcFromServer = getSegmentCrcFromServer(realtimeTableName, segmentName, uri.toString());
```

Comment:
Currently we read crc and ask server to upload as 2 separate steps. A more ideal solution would be to enhance the server upload API to return both segment location (i.e. `tempSegmentDownloadUrl`) and also crc. The response can be a json wrapping more info other than the download url.

Reply:
> enhance the server upload API to return both segment location (i.e. tempSegmentDownloadUrl) and also crc. 

This change would require implementing a new server API, as users might be directly utilizing the existing API, making the modification backward-incompatible. If introducing a new API is acceptable, I can proceed with its implementation.

Reply:
A new API is okay since the response format is changing. To keep it backward compatible, controller can first try the new API, and fallback to the old API if new API returns error.

Reply:
Sgtm! Updated accordingly.